salt/modules/file.py

```diff
@@ -1728,11 +1735,11 @@ def line(path, content=None, match=None, mode=None, location=None,
         log.warning('Cannot find text to {0}. File \'{1}\' is empty.'.format(mode, path))
         body = ''
     elif mode == 'delete':
-        body = os.linesep.join([line for line in body.split(os.linesep) if line.find(match) < 0])
+        body = os.linesep.join([line for line in _split(body) if line.find(match) < 0])
```

Comment:
Joining with os.linesep and splitting by something else will lead to all the line breaks replaced in the file...

Reply:
Seems legit to me, to output file with line endings proper to OS.
Converting '\n' to os.linesep is default python behavior for not binary writes.